parser_library/src/context/hlasm_context.cpp

```diff
@@ -144,117 +164,111 @@ void hlasm_context::add_system_vars_to_scope(code_scope& scope)
                     tmp = *it->this_macro->id;
                 else
                     tmp = "OPEN CODE";
-                data.push_back(std::make_unique<macro_param_data_single>(std::move(tmp)));
+                data.push_back(std::move(tmp));
             }
 
-            macro_data_ptr mac_data = std::make_unique<macro_param_data_composite>(std::move(data));
-
-            sys_sym_ptr var = std::make_shared<system_variable_sysmac>(SYSMAC, std::move(mac_data), false);
-
-            scope.system_variables.insert({ SYSMAC, std::move(var) });
+            scope.system_variables.insert(
+                create_system_variable<system_variable_sysmac>(ids(), "SYSMAC", std::move(data), false));
         }
     }
 }
 
-void hlasm_context::add_global_system_vars(code_scope& scope)
+void hlasm_context::add_global_system_var_to_scope(id_storage& ids, std::string name, code_scope& scope) const
 {
-    auto SYSDATC = ids().add("SYSDATC");
-    auto SYSDATE = ids().add("SYSDATE");
-    auto SYSTIME = ids().add("SYSTIME");
-    auto SYSPARM = ids().add("SYSPARM");
-    auto SYSOPT_RENT = ids().add("SYSOPT_RENT");
-    auto SYSTEM_ID = ids().add("SYSTEM_ID");
+    auto id = ids.find(name);
```

Comment:
There is no benefit to trying to find it first...
```suggestion
    auto id = ids.add(name);
```